aws_lambda_powertools/event_handler/appsync.py

```diff
@@ -1,21 +1,78 @@
 import logging
-from typing import Any, Callable, Optional, Type, TypeVar
+from abc import ABC, abstractmethod
+from itertools import groupby
+from typing import Any, Callable, Dict, List, Optional, Type, Union
 
 from aws_lambda_powertools.utilities.data_classes import AppSyncResolverEvent
 from aws_lambda_powertools.utilities.typing import LambdaContext
 
 logger = logging.getLogger(__name__)
 
-AppSyncResolverEventT = TypeVar("AppSyncResolverEventT", bound=AppSyncResolverEvent)
 
+class RouterContext:
+    def __init__(self):
+        self._context = {}
+
+    @property
+    def context(self) -> Dict[str, Any]:
+        return self._context
+
+    @context.setter
+    def context(self, additional_context: Dict[str, Any]) -> None:
+        """Append key=value data as routing context"""
+        self._context.update(**additional_context)
+
+    @context.deleter
+    def context(self):
+        """Resets routing context"""
+        self._context.clear()
+
+
+class IResolverRegistry(ABC):
```

Comment:
Perhaps you were programming in another language and forgot this?

We don't do that here ;-)

Reply:
Eh, I used different convention I'm using in different project ;-) Python doesn't have agreed convention for naming interfaces ( at least there is no mention of this in PEP8) and thats what happens :-) Anyway, will adjust it to our own convention. Thx!

Reply:
hey! Pushed one more commit to the branch. It address 3 things:
* Refactor interface naming
* Decided to change dependencies between Router and AppSyncResolver. Since it seems they are heavy dependent, have the same public interfaces it makes sense to use inheritance here. This way I dont need to copy the same interfaces between two classes as before
* Simplifying `_call_batch_resolver` method and remove groupby
cc @leandrodamascena